docs/src/main/asciidoc/spanner.adoc

```diff
@@ -317,8 +317,11 @@ CREATE TABLE items (
 ) PRIMARY KEY (grandParentId, parentId, id)
 ----
 
-Note that embedded properties' column names must all be unique.
+Note that the following restrictions apply when you use embedded objects:
 
+- embedded properties' column names must all be unique.
+- embedded properties must be mutable; otherwise you'll get an error, such as `SpannerDataException: Column not found: parent`.
+  Therefore, you cannot use Kotlin's data class, which is immutable, to hold an embedded property.
```

Comment:
This is not accurate. Kotlin's data class can have mutable field with 'var'. But the following definition (`@Embedded var entityHistory`) failed too:

```
@Table(name = "Requests")
data class Request(
    @PrimaryKey val requestId: String,
    val payload: String,
    @Embedded var entityHistory: EntityHistory? = null
)
```

What's the better description?

Reply:
Can we rephrase somehow that the constructor should not contain embedded properties?

It seems that to use Kotlin data class with `@Embedded`, we can do something like:

```kt
@Table(name = "Requests")
data class Request(
    @PrimaryKey val requestId: String,
    val payload: String,
) {
    @Embedded lateinit var entityHistory: EntityHistory
}
```

or

```kt
@Table(name = "Requests")
data class Request(
    @PrimaryKey val requestId: String,
    val payload: String,
) {
    @Embedded var entityHistory: EntityHistory? = null
}
```

Reply:
How about "Embedded properties must not be passed through constructor"? This would apply to both Java and Kotlin.

Reply:
@elefeint @gaeshi Thank you for feedback. Updated.